src/main/java/Storage.java

```diff
@@ -0,0 +1,201 @@
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Scanner;
+
+/**
+ * A Storage class that deals with loading tasks from the file and saving tasks in the file.
+ */
+public class Storage {
+
+    private File file;
+
+    /**
+     * Creates a Storage object.
+     *
+     * @param filePath A string that represents the directory of a file.
+     */
+    public Storage(String filePath) {
+        file = new File(filePath);
+    }
+
+    /**
+     * Updates the task in the file to be marked as done.
+     *
+     * @param number An int representing the task number.
+     */
+    public void updateDone(int number) {
+        try {
+            String s = "";
+            Scanner scanFile = new Scanner(file);
+            for (int i = 1; i <= TaskList.size(); i++) {
+                if (i == number) {
+                    if (i == TaskList.size()) {
+                        s = s + scanFile.nextLine().replaceFirst("0", "1");
```

Comment:
What if a task had already been marked as done, and the task description has a 0 in it?

_Suggestion \:_
- Either split the nextLine by the pipeline character seperator '|' and edit the second element of the array
- Or change the done command not to execute if the task was already marked as done!